nautobot_golden_config/views.py

```diff
@@ -258,40 +264,79 @@ def get_required_permission(self):
         """Manually set permission when not tied to a model for config details."""
         return "nautobot_golden_config.view_goldenconfig"
 
-    def get(self, request, pk, config_type):  # pylint: disable=invalid-name,too-many-branches
+    def get(
+        self, request, pk, config_type
+    ):  # pylint: disable=invalid-name,too-many-branches,too-many-locals,too-many-statements
         """Read request into a view of a single device."""
+
+        def diff_structured_data(backup_data, intended_data):
+            """Utility function to provide `Unix Diff` between two JSON snippets."""
+            backup_yaml = yaml.safe_dump(json.loads(backup_data))
+            intend_yaml = yaml.safe_dump(json.loads(intended_data))
+
+            for line in difflib.unified_diff(backup_yaml.splitlines(), intend_yaml.splitlines(), lineterm=""):
+                yield line
+
         device = Device.objects.get(pk=pk)
         config_details = models.GoldenConfig.objects.filter(device=device).first()
+        if not config_details and config_type == "json_compliance":
+            # Create the GoldenConfig object for the device only for JSON compliance.
+            config_details = models.GoldenConfig.objects.create(device=device)
         structure_format = "json"
         if not config_details:
             output = ""
         elif config_type == "backup":
             output = config_details.backup_config
         elif config_type == "intended":
             output = config_details.intended_config
-        elif config_type == "compliance":
-            output = config_details.compliance_config
-            if config_details.backup_last_success_date:
-                backup_date = str(config_details.backup_last_success_date.strftime("%b %d %Y"))
-            else:
-                backup_date = datetime.now().strftime("%b %d %Y")
-            if config_details.intended_last_success_date:
-                intended_date = str(config_details.intended_last_success_date.strftime("%b %d %Y"))
+        elif "compliance" in config_type:
+            if config_type == "compliance":
```

Comment:
Can you add a comment on these conditions to explain the human side of what is going on?